crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1714,29 +1703,41 @@ fn field_mismatch_message(old_field: &Field, new_field: &Field) -> (Declarations
         (true, true) => (
             Declarations::FieldMismatch,
             format!(
-                "Mismatched field {}, expected {}.",
-                new_field_full, old_field_full
+                "Mismatched field '{new_field_full}'{new_pos}, expected '{old_field_full}'{}.",
+                if new_pos.is_empty() {
+                    old_pos
+                } else {
+                    "".to_string()
+                }
             ),
         ),
         (true, false) => (
             Declarations::FieldMismatch,
             format!(
                 "Mismatched field name {}, expected {}.",
```

Comment:
When you have a positional field, this is going to say "Mismatched field name positional field", probably dropping the "name" is sufficient?